core/src/tile/mapTile.cpp

```diff
@@ -47,13 +44,15 @@ void MapTile::draw(const Style& _style, const View& _view) {
 
         std::shared_ptr<ShaderProgram> shader = _style.getShaderProgram();
 
-        glm::dmat4 modelViewProjMatrix = _view.getViewProjectionMatrix() * m_modelMatrix;
+        // Apply tile-view translation to the model matrix
+        const auto& viewOrigin = _view.getPosition();
+        m_modelMatrix[3][0] = m_tileOrigin.x - viewOrigin.x;
```

Comment:
Technically yes, but this is essentially the same as our current implementation (we converted to floats [here](https://github.com/tangrams/tangram-es/pull/48/files#diff-1563124aa3d1828325c97ec833bb3c88L54)). The idea with this is that these translation values are individually too large to represent well with floats, but their difference is perfectly fine to keep as a float.